### ISSUE TITLE
Don't start the debugger when running `rails console` or `pry` command (as opposed to using binding.pry in a file)

### ISSUE THREAD

**Issue by TylerRick** [garmoshka-mo/pry-moves#9]
With `pry-rails` gem installed in a rails app, when you start the `rails console`, you get a normal pry session like this:
```
⟫ rails c
Loading development environment (Rails 5.2.3)
[1] pry(main)> 
```

But as soon as you also add the `pry-moves` gem to the project, you get this as your starting point instead:

```
⟫ rails c
Loading development environment (Rails 5.2.3)

Frame: 0/30 method
From: pry-moves/lib/pry-moves/pry_ext.rb:14 Pry

     4: def start_with_pry_nav(target = TOPLEVEL_BINDING, options = {})
     5:   old_options = options.reject { |k, _| k == :pry_remote }
     6: 
     7:   if target.is_a?(Binding) && PryMoves.check_file_context(target)
     8:     # Wrap the tracer around the usual Pry.start
     9:     PryMoves::PryWrapper.new(target, options).run do
    10:       start_without_pry_nav(target, old_options)
    11:     end
    12:   else
    13:     # No need for the tracer unless we have a file context to step through
 => 14:     start_without_pry_nav(target, old_options)
    15:   end
    16: end

[1] pry(Pry)> 
```

Pry seems to still work for the most part, but it adds unwanted noise and is surprising to see a debugger context where I'm just expecting a normal REPL.

By the way, I get the same thing when I just run `bundle exec pry`, so it doesn't even seem like it's a Rails-specific problem.

It seems like the pry-moves debugger should not be enabled in the context of a rails console — until and unless some code _in a file_ explicitly calls `binding.pry`.

Ironically, the comment line for the branch that it appears to be taking even implies that it correctly detected that we're _not_ in a file context and _don't_ need the tracer:
```
    13:     # No need for the tracer unless we have a file context to step through
 => 14:     start_without_pry_nav(target, old_options)
```
Yet we still have the tracer.

An additional side effect is that if you happen to execute the `s` command from this context, it breaks...

```
[1] pry(Pry)> s
UncaughtThrowError: uncaught throw :breakout_nav
from pry-moves/lib/pry-moves/commands.rb:67:in `throw'
bundler: failed to load command: pry (/home/tyler/.gem/ruby/2.6.0/bin/pry)
NoMethodError: private method `eval' called for nil:NilClass
```

I'm not even sure how to fix this because I don't understand what's causing it to get to that code.

**Comment by nilbus** [comment]
I'd also like to have this resolved. I'm hoping to look into it soon.

I confirmed that this branch of code where the `PryMoves::PryWrapper` wraps `start_without_pry_nav` is not being executed.

```ruby
     7:   if target.is_a?(Binding) && PryMoves.check_file_context(target)
     8:     # Wrap the tracer around the usual Pry.start
     9:     PryMoves::PryWrapper.new(target, options).run do
    10:       start_without_pry_nav(target, old_options)
    11:     end
```

@garmoshka-mo Please let me know if you have any insight into why `whereami` might be triggered every time, as described in this issue.

**Comment by garmoshka-mo** [comment]
@TylerRick @nilbus Thank you for reported issue and detailed description
To debug this, please go to: `lib/pry-stack_explorer/when_started_hook.rb` and disable `bindings = remove_internal_frames(bindings)`

Then you will be able to see details of things which happen, namely `:when_started` hook triggers `:save_caller_bindings` at `console.start`, I assume this changes target and later, when `whereami` triggered by Pry (it's triggered every time, when console printed) - internal_binding? returns false therefore information about location printed

I assume one of possible solutions could be to conditionally disable `save_caller_bindings` for console start

**Comment by nilbus** [comment]
> Then you will be able to see details of things which happen, namely `:when_started` hook triggers `:save_caller_bindings` at `console.start`, I assume this changes target and later, when `whereami` triggered by Pry (it's triggered every time, when console printed) - internal_binding? returns false therefore information about location printed

💯 Yes, that's exactly what is happening.

> I assume one of possible solutions could be to conditionally disable `save_caller_bindings` for console start

I think this would be the best solution. So far, I haven't found a clean way to differentiate between console start and a call to `pry` from the source, other than trying to infer it from the call stack. I'm searching for a better way.

Another approach could be to examine the call stack to determine whether pry was started via `Pry.start` or `binding.pry`. Consoles generally all start via `Pry.start` (checked rails, guard, and gems generated via bundler), and debugging sessions generally start via `Binding#pry` or `Object#pry`.

I'm still working on this, but let me know if you have any other thoughts on these approaches.

**Comment by garmoshka-mo** [comment]
Fixed in v0.1.12

**Comment by nilbus** [comment]
Thanks!
